File: python/Book-Discounts-Refactoring-Kata/bookstore/discount_voucher.py
from bookstore.cartable import CartAble
from bookstore.cart import Cart


class DiscountVoucher(CartAble):

    def __init__(self, percentage: int):
        self.percentage = percentage

    def put_into_my_cart(self, cart: Cart):
        cart.add(f"Discount voucher for {self.percentage}% on your next order")

File: cpp/src/DiscountVoucher.cpp
//
// Created by mille on 28-Sep-22.
//

#include "DiscountVoucher.h"

namespace bookstore {
  DiscountVoucher::DiscountVoucher(int _percentage) {
    percentage = _percentage;
  }

  void DiscountVoucher::putIntoMyCart(Cart& cart) {
    cart.add("Discount voucher for " + std::to_string(percentage) + "% on your next order");
  }
} // bookstore